src/components/MapView/Analyser/index.tsx

```diff
@@ -0,0 +1,378 @@
+import {
+  AnalysisDispatchParams,
+  analysisResultSelector,
+  clearAnalysisResult,
+  isAnalysisLayerActiveSelector,
+  isAnalysisLoadingSelector,
+  requestAndStoreAnalysis,
+  setIsMapLayerActive,
+} from '../../../context/analysisResultStateSlice';
+import AnalysisTable from './AnalysisTable';
+import LayerDropdown from '../../../utils/LayerDropdown';
+
+const layers = Object.values(LayerDefinitions);
+
+const boundaryLayer = getBoundaryLayerSingleton();
+
+function Analyser({ classes }: AnalyserProps) {
+  const dispatch = useDispatch();
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const availableDates = useSelector(availableDatesSelector);
+  const analysisResult = useSelector(analysisResultSelector);
+
+  const isAnalysisLoading = useSelector(isAnalysisLoadingSelector);
+  const isMapLayerActive = useSelector(isAnalysisLayerActiveSelector);
+
+  const [isAnalyserFormOpen, setIsAnalyserFormOpen] = useState(false);
+  const [isTableViewOpen, setIsTableViewOpen] = useState(true);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState(
+    (layers.find(layer => layer.type === 'wms') as WMSLayerProps).id,
+  );
+  const [statistic, setStatistic] = useState(AggregationOperations.Mean);
+  const [baselineLayerId, setBaselineLayerId] = useState(
+    (layers.find(layer => layer.type === 'nso') as NSOLayerProps).id,
+  );
+  const [selectedDate, setSelectedDate] = useState<number | null>(null);
+
+  const [belowThreshold, setBelowThreshold] = useState('');
+  const [aboveThreshold, setAboveThreshold] = useState('');
+  const [thresholdError, setThresholdError] = useState<string | null>(null);
+
+  // enforce errors.
+  useEffect(() => {
+    const belowThresholdValue = parseFloat(belowThreshold);
+    const aboveThresholdValue = parseFloat(aboveThreshold);
+    if (belowThresholdValue > aboveThresholdValue) {
+      setThresholdError('Min threshold is larger than Max!');
+    } else setThresholdError(null);
+  }, [belowThreshold, aboveThreshold]);
+
+  // set default date after dates finish loading and when hazard layer changes
+  useEffect(() => {
+    const dates =
+      availableDates[
+        (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
+      ];
+    if (!dates || dates.length === 0) {
+      setSelectedDate(null);
+    } else setSelectedDate(dates[dates.length - 1]);
+  }, [availableDates, hazardLayerId]);
+
+  const onOptionChange = (setterFunc: (val: any) => void) => (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setterFunc((event.target as HTMLInputElement).value);
+  };
+
+  const adminBoundariesExtent = useMemo(() => {
+    if (!boundaryLayerData)
+      // not loaded yet. Should be loaded in MapView
+      return null;
+    return bbox(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
```

Comment:
We indeed must cast it since `bbox` also supports calculating bboxes for 3D features. Hence, bbox has a typing for both 3D and 2D number arrays (length 4 and 6 respectively). Since we are definitely working with 2D, we need to let TS know we are expecting the 2D version out of `bbox`